lib/routes/routePUT.js

```diff
@@ -33,6 +36,40 @@ export default function routePUT(request, response, log, utapi) {
                     return routesUtils.responseNoBody(err, null, response, 200,
                         log);
                 });
+            } else if (request.query.versioning !== undefined) {
+                if (request.post === '') {
+                    log.debug('request xml is missing');
+                    return routesUtils.responseNoBody(
+                        errors.MalformedXML, null, response, null, log);
+                }
+                const xmlToParse = request.post;
+                return parseString(xmlToParse, (err, result) => {
```

Comment:
Is there a reason to do the parsing here in the routes?  I'd prefer it in the api if possible, but it might be the right way to do things for error compatibility.  @nicolas2bert , do you have a view?